src/App/Fossa/Analyze/GraphBuilder.hs

```diff
@@ -16,26 +16,22 @@ import Control.Carrier.State.Strict
 import qualified App.Fossa.Analyze.Graph as G
 import DepTypes
 
-data GraphBuilder m k
-  = AddNode Dependency (G.DepRef -> m k)
-  | AddEdge G.DepRef G.DepRef (m k)
-  | AddDirect G.DepRef (m k)
-  deriving Generic1
-
-instance HFunctor GraphBuilder
-instance Effect GraphBuilder
+data GraphBuilder m k where
+  AddNode :: Dependency -> GraphBuilder m G.DepRef
+  AddEdge :: G.DepRef -> G.DepRef -> GraphBuilder m ()
+  AddDirect :: G.DepRef -> GraphBuilder m ()
 
 -- | Add a node to the graph. See 'G.addNode'
 addNode :: Has GraphBuilder sig m => Dependency -> m G.DepRef
-addNode node = send (AddNode node pure)
```

Comment:
The `pure` here was another common confusion when writing constructors for an effect